source/AMDSBufferGroup.cpp

```diff
@@ -66,11 +73,28 @@ void AMDSBufferGroup::populateData(AMDSClientStartTimePlusCountDataRequest* clie
 			AMDSFlatArray oneFlatArray;
 			dataHolder->data(&oneFlatArray);
 			clientStartTimePlusCountDataRequest->appendData(dataHolder);
-//			request->histogramData()->append(dataHolders_[iCurrent]);
 		}
 	}
 }
 
+void AMDSBufferGroup::populateData(AMDSClientRelativeCountPlusCountDataRequest* clientDataRequest)
+{
+	int relativeCount = clientDataRequest->relativeCount();
+	int count = clientDataRequest->count();
+
+	int startIndex = dataHolders_.count() - 1 - relativeCount;
+	int endIndex = dataHolders_.count() - 1 - relativeCount + count;
+
+	startIndex = (startIndex < 0 ? 0 : startIndex);
```

Comment:
There are handy functions from Qt that you can use rather than making these types of conditionals.  For instance you could replace these two lines with:

```
startIndex = qMax(0, startIndex);
endIndex = qMin(endIndex, dataHolders_.count()-1);
```

Although functionally equivalent, I think it would be easier to read using the above suggestion.